tests/test_object_detection.py

```diff
@@ -601,3 +604,76 @@ def test_visualize(monkeypatch, generate_single_image_file):
         },
         overlay=False,
     )
+
+
+def test_has_labels_overlap():
+    bboxes = np.array(
+        [
+            [359.0, 146.0, 472.0, 360.0],
+            [340.0, 22.0, 494.0, 323.0],
+            [472.0, 173.0, 508.0, 221.0],
+            [486.0, 183.0, 517.0, 218.0],
+            [359.0, 144.0, 470.0, 358.0],
+        ]
+    )
+    label_classes = [0, 1, 2, 3, 2]
+    is_overlaps = _has_overlap(bboxes, label_classes)
+    expected_res = np.array([True, False, False, False, True])
+    assert np.array_equal(is_overlaps, expected_res)
+
+
+@pytest.mark.parametrize("overlapping_label_check", [True, False])
+def test_swap_overlap_labels(overlapping_label_check):
+    prediction = predictions[3].copy()
+    label = labels[3].copy()
+    label["bboxes"] = np.append(label["bboxes"], [label["bboxes"][-1]], axis=0)
+    label["labels"] = np.append(label["labels"], (label["labels"][-1] + 1) % 10)
+    score = get_label_quality_scores(
+        [label], [prediction], overlapping_label_check=overlapping_label_check
+    )[0]
+    if overlapping_label_check:
+        assert score < 0.06
+    else:
+        assert score < 0.08
+
+
+@pytest.mark.parametrize("overlapping_label_check", [True, False])
+def test_swap_only_overlap_labels(overlapping_label_check):
+    prediction = predictions[3].copy()
+    label = labels[3].copy()
+    label["bboxes"] = np.append(label["bboxes"], [label["bboxes"][-1]], axis=0)
+    label["labels"] = np.append(label["labels"], (label["labels"][-1] + 1) % 10)
+    score = compute_swap_box_scores(
+        [label], [prediction], overlapping_label_check=overlapping_label_check
+    )[0]
+    if overlapping_label_check:
+        assert np.allclose(score, np.array([0.88, 1.0, 0.95, 0.96, 1.0, 0.0, 0.0]), atol=1e-2)
+    else:
+        assert np.allclose(score, np.array([0.88, 1.0, 0.95, 0.96, 1.0, 0.88, 0.0]), atol=1e-2)
+
+
+def test_badloc_high_low_probability_threshold():
+    prediction = predictions[3].copy()
+    label = labels[3].copy()
+    label["bboxes"] = np.append(label["bboxes"], [label["bboxes"][-1]], axis=0)
+    label["labels"] = np.append(label["labels"], (label["labels"][-1] + 1) % 10)
+    score = compute_badloc_box_scores([label], [prediction], low_probability_threshold=1.0)[0]
+    assert np.allclose(score, np.ones_like(score), atol=1e-2)
+
+
+def test_overlooked_high_high_probability_threshold():
```

Comment:
Why are these functions called "test_swap_high_high" or "test_badloc_high_low". Shouldn't it just be "test_badloc_low_probability_threshold()" and "test_swap_high_probability_threshold()"?